app/services/katello/pulp3/module_stream.rb

```diff
@@ -25,11 +25,23 @@ def self.build_stream_rpms(katello_id, package_hrefs)
       end
 
       def self.build_artifacts(katello_id, artifacts_json)
-        return [] if artifacts_json.nil?
-        artifacts = artifacts_json.map do |name|
-          {name: name,
-           module_stream_id: katello_id}
+        return [] if artifacts_json.empty?
+
+        if artifacts_json.is_a?(::Hash)
+          artifacts = []
+          artifacts_json.each do |_type, artifact_names|
+            artifacts << artifact_names.collect do |name|
```

Comment:
Only question I have is can the same artifact names be in multiple types ?

Reply:
It should not since before, Pulp was simply displaying the "rpms" type without actually including the type itself. Other types seem to be things more like "description" rather than more types of content. I'll make another issue to fully support their new artifacts hash.

Reply:
Also I'm testing syncing RHEL 8 AppStream now.

Reply:
RHEL 8 AppStream works! That's the golden standard so I'd say we're all good.

Reply:
![image](https://user-images.githubusercontent.com/14796566/220201042-1928badc-f2d3-4b48-97c7-488c78b99f28.png)

Reply:
![image](https://user-images.githubusercontent.com/14796566/220201076-0622a0ad-8b70-4a41-ae09-87edbfd3ea5e.png)